src/test/java/com/commercetools/sync/products/ProductSyncMockUtils.java

```diff
@@ -119,6 +121,27 @@ public static ProductDraftBuilder createProductDraftBuilder(@Nonnull final Strin
             .publish(productFromJson.getMasterData().isPublished());
     }
 
+    /**
+     * Given a {@link Set} of {@link Category} {@link ResourceIdentifier}, this method returns an instance of
+     * {@link CategoryOrderHints} containing a {@link Map}, in which each entry has category id from the supplied
+     * {@link Set} as a key and a random categoryOrderHint which is a {@link String} containing a random double value
+     * between 0 and 1 (exclusive).
+     *
+     * <p>Note: The random double value is generated by the {@link ThreadLocalRandom#current()} nextDouble method.
+     *
+     * @param categoryResourceIdentifiers set of resource identifiers of categories to build categoryOrderHints for.
+     * @return an instance of {@link CategoryOrderHints} containing a categoryOrderHint for each category in the
+     *         supplied set of category resource identifiers.
+     */
+    public static CategoryOrderHints createRandomCategoryOrderHints(
+        @Nonnull final Set<ResourceIdentifier<Category>> categoryResourceIdentifiers) {
+
+        final List<Reference<Category>> references = categoryResourceIdentifiers
+            .stream()
+            .map(categoryResourceIdentifier -> Category.referenceOfId(categoryResourceIdentifier.getId()))
+            .collect(toList());
+        return createRandomCategoryOrderHints(references);
+    }
 
     /**
      * Given a {@link List} of {@link Category}, this method returns an instance of {@link CategoryOrderHints}
```

Comment:
``Given a {@link Set} of {@link Category} {@link Reference}``

Reply:
why? It is still a list here :)

Reply:
probably I was reading the other one, methods have the same name. 🙈